src/renderer/dx/DxFontRenderData.h

```diff
@@ -75,26 +75,30 @@ namespace Microsoft::Console::Render
         [[nodiscard]] static HRESULT STDMETHODCALLTYPE s_CalculateBoxEffect(IDWriteTextFormat* format, size_t widthPixels, IDWriteFontFace1* face, float fontScale, IBoxDrawingEffect** effect) noexcept;
 
     private:
-        void _BuildFontRenderData(const FontInfoDesired& desired, FontInfo& actual, const int dpi);
-        Microsoft::WRL::ComPtr<IDWriteTextFormat> _BuildTextFormat(const DxFontInfo fontInfo, const std::wstring_view localeName);
+        using FontAttributeMapKey = uint32_t;
 
-        ::Microsoft::WRL::ComPtr<IDWriteFactory1> _dwriteFactory;
+        // We use this to identify font variants with different attributes.
+        static FontAttributeMapKey _ToMapKey(DWRITE_FONT_WEIGHT weight, DWRITE_FONT_STYLE style, DWRITE_FONT_STRETCH stretch) noexcept
+        {
+            return (weight << 16) | (style << 8) | stretch;
+        };
 
-        ::Microsoft::WRL::ComPtr<IDWriteTextAnalyzer1> _dwriteTextAnalyzer;
+        void _BuildFontRenderData(const FontInfoDesired& desired, FontInfo& actual, const int dpi);
+        Microsoft::WRL::ComPtr<IDWriteTextFormat> _BuildTextFormat(const DxFontInfo fontInfo, const std::wstring_view localeName);
 
-        std::unordered_map<DxFontInfo, ::Microsoft::WRL::ComPtr<IDWriteTextFormat>> _textFormatMap;
-        std::unordered_map<DxFontInfo, ::Microsoft::WRL::ComPtr<IDWriteFontFace1>> _fontFaceMap;
+        std::unordered_map<FontAttributeMapKey, ::Microsoft::WRL::ComPtr<IDWriteTextFormat>> _textFormatMap;
+        std::unordered_map<FontAttributeMapKey, ::Microsoft::WRL::ComPtr<IDWriteFontFace1>> _fontFaceMap;

```

Comment:
I believe we can get rid of these hashmaps entirely:
As far as I can see we currently don't actually support dynamic font stretches and don't plan to add them in the foreseeable future either. So we can drop those for now, allowing us to simplify our code.

With "stretch" dropped, we currently only support two kinds of bold (on/off) and two kinds of italic (on/off). It's unlikely that this will change in the foreseeable future either.
As I personally generally don't think it's a good idea to plan for the unknown (in programming), I'd like to suggest to change these line to this in a different PR:
```cpp
::Microsoft::WRL::ComPtr<IDWriteTextFormat>> _textFormatMap[2][2];
::Microsoft::WRL::ComPtr<IDWriteFontFace1>> _fontFaceMap[2][2];
```

If we then have for instance `TextFormatWithAttribute(bool bold, bool italic)` we can write `_textFormatMap[bold][italic]` for 0-overhead access to the format map. No allocations, or complex hashing involved. 👍

Reply:
Agreed! This would be an excellent simplification.

I knew we initially designed it for font stretch, but here's what I think now: we can change it if we ever need to easily support Stretch. After all, let's go with the simplest implementation that works for any given problem unless we truly do need a complex one. :smile:

Reply:
I was trying to reply using email. But that email was not sent successfully. 

Just want to say I am also OK with the simple implementation. It’s a good idea👍